spec/forms/openid_connect_authorize_form_spec.rb

```diff
@@ -356,6 +356,65 @@
     end
   end
 
+  describe '#aal' do
+    context 'when AAL1 passed' do
+      before do
+        IdentityConfig.store.valid_authn_contexts.push(Saml::Idp::Constants::AAL1_AUTHN_CONTEXT_CLASSREF)
+      end
+
+      after do
+        IdentityConfig.store.valid_authn_contexts.pop
+      end
+
+      let(:acr_values) { Saml::Idp::Constants::AAL1_AUTHN_CONTEXT_CLASSREF }
+
+      it 'returns 1' do
+        expect(form.aal).to eq(1)
+      end
+    end
+
+    context 'when AAL2 passed' do
+      let(:acr_values) { Saml::Idp::Constants::AAL2_AUTHN_CONTEXT_CLASSREF }
```

Comment:
**suggestion:** can we pass in something that combines IAL and AAL specifications separated by a space (since IAL is always required) to make this more realistic? e.g.

```ruby
"#{Saml::Idp::Constants::IAL1_AUTHN_CONTEXT_CLASSREF} #{Saml::Idp::Constants::AAL2_AUTHN_CONTEXT_CLASSREF}"
```

Reply:
Test added for ial and aal in the acr together. Unrelated, autocorrect hated most of that sentence.

Reply:
yeah we really do a number on autocorrect don't we 😂